### PR TITLE
Update onbuild to not 'run'

### PR DESCRIPTION
**Pull request by richlander**
/cc @MichaelSimons @kendrahavens

### CODE REVIEW

**Comment by MichaelSimons** [comment]
IMO we should not be making this change without revving the docker tag.  This is a behavior change and considered a breaking change.  Since our version number is tied to the CLI, that means we would have to wait until we pickup preview3.  This can be made in dotnet-nightly now however.

File: 1.0.0-preview2/debian/onbuild/Dockerfile [hunk 1]
```diff
@@ -3,7 +3,9 @@ FROM microsoft/dotnet:1.0.0-preview2-sdk
 RUN mkdir -p /dotnetapp
```
Comment (MichaelSimons) [inline 1.0.0-preview2/debian/onbuild/Dockerfile]: I think now would be a good time to remove this to simplify the Dockerfile.  Additionally all COPY destinations should be changed to `.` or `./`.

File: 1.0.0-preview2/debian/onbuild/Dockerfile [hunk 2]
```diff
@@ -3,7 +3,9 @@ FROM microsoft/dotnet:1.0.0-preview2-sdk
...
 WORKDIR /dotnetapp
 
-ENTRYPOINT ["dotnet", "run"]
+ENTRYPOINT ["dotnet", "out/dotnetapp"]
```
Comment (MichaelSimons) [inline 1.0.0-preview2/debian/onbuild/Dockerfile]: It looks like this broke CI.  I think it is missing `.dll`.  The same comment applies to the nanoserver version

Comment (MichaelSimons) [inline 1.0.0-preview2/debian/onbuild/Dockerfile]: This is not going to work for named projects.  By that I mean apps that specify a name in the `project.json`.  When a name is specified, dotnet creates the resulting assembly with that name.  We would need to retrieve this name dynamically.  I am not aware of any functionality the CLI provides for this so we would likely have to write our own logic.

**Comment by richlander** [comment]
I'll close this PR and make it in dotnet-nightly. Good point on compat.

Ya, you can see I didn't actually test it. I wanted to use it to have the conversation. I'm happy with the answer.

For onbuild, I'm happy to give up on the name property. The general improvement seems to be significant enough to give up on that scenario. Basically, you've grown out of on-build at that point!

Cool?

**Pull request closed without merging**